### PR TITLE
Vslt/vslt 365 download

### PR DESCRIPTION
**Pull request by xwa130saucer**
We are able to download 4.5.4 (could be customized) when user install screener-runner.

All changes in `package-lock.json` is introduced when I ran `npm install --save sauce-connect-launcher`.

### CODE REVIEW

File: package.json
Comment (loyalchow) [inline package.json]: Let's use `~` or pin this version to be safe

File: package.json
Comment (vrunoa) [inline package.json]: @xwa130saucer check the original post-install script; https://github.com/bermi/sauce-connect-launcher/blob/master/package.json#L21
This is probably done this way to support different node versions, we should try to handle this too.

Comment (vrunoa) [reply package.json]: @xwa130saucer what do you think of this approach as an alternative:
```
SAUCE_CONNECT_DOWNLOAD_ON_INSTALL=true SAUCE_CONNECT_VERSION=4.5.4 npm run postinstall --prefix ./node_modules/sauce-connect-launcher
```

**Comment by vrunoa** [comment]
@loyalchow @xwa130saucer sauce-connect-launcher says that it should work with node >=4 but it isnt working o node 4,5. At least on my machine. Lets make sure it works for the versions we support. 

Fails to unzip the archive, we should review the supported versions. 
```
Failed to download sauce connect binary: [Error: Couldn't unpack archive: ERROR Unzipping file: Buffer.alloc is not a function]
sauce-connect-launcher will attempt to re-download next time it is run.
```

https://github.com/bermi/sauce-connect-launcher/blob/master/package.json#L48-L50

File: package.json
Comment (xwa130saucer) [reply package.json]: Just clarify the difference of `~` and `^` here to make sure this is what we really want to do:
`~`: we limit the change only to `1.3.x`
`^`: we limit the change to `1.x`
Have updated.

**Comment by loyalchow** [comment]
@vrunoa @xwa130saucer For the node versions, does sauce-connect-launcher work on node >= 6?
If yes, then let's change screener-runner to support node >= 6 only (screener-storybook already only supports v6+)

File: package.json
Comment (xwa130saucer) [reply package.json]: > @xwa130saucer check the original post-install script; https://github.com/bermi/sauce-connect-launcher/blob/master/package.json#L21
> This is probably done this way to support different node versions, we should try to handle this too.

From my understanding, the later part `|| nodejs scripts/install.js` is a guarantee that the system does not have `node` instead of `nodejs` command line. But I don't understand how they handle version... I'll add that part in case of the host does not have `node` command line. And could you please clarify the version handle part? didn't quite get it. 


> @xwa130saucer what do you think of this approach as an alternative:
> 
> ```
> SAUCE_CONNECT_DOWNLOAD_ON_INSTALL=true SAUCE_CONNECT_VERSION=4.5.4 npm run postinstall --prefix ./node_modules/sauce-connect-launcher
> ```

I think this command line will fire the `sauce-connect-launcher` which is not what we want to do right? We are just downloading the sauce connect client at this phase. 
Also, from my understanding, the postinsall is a hook that is automatically called after the package is installed.

**Comment by xwa130saucer** [comment]
> @loyalchow @xwa130saucer sauce-connect-launcher says that it should work with node >=4 but it isnt working o node 4,5. At least on my machine. Lets make sure it works for the versions we support.
> 
> Fails to unzip the archive, we should review the supported versions.
> 
> ```
> Failed to download sauce connect binary: [Error: Couldn't unpack archive: ERROR Unzipping file: Buffer.alloc is not a function]
> sauce-connect-launcher will attempt to re-download next time it is run.
> ```
> 
> https://github.com/bermi/sauce-connect-launcher/blob/master/package.json#L48-L50

I just tested on my machine, I used node `4.5.0` but didn't encounter your issue. It downloaded properly. However, I met this issue when I ran my test:
```
/Users/xinyiwang/Xinyi/projects/integrate-with-sc/screener-runner-example/node_modules/screener-runner/src/tunnel.js:5
exports.connect = function(host, token, tries = 0) {
                                              ^
SyntaxError: Unexpected token =
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/Users/xinyiwang/Xinyi/projects/integrate-with-sc/screener-runner-example/node_modules/screener-runner/src/runner.js:4:14)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/Users/xinyiwang/Xinyi/projects/integrate-with-sc/screener-runner-example/node_modules/screener-runner/src/cli.js:6:14)
    at Module._compile (module.js:409:26)
```
It happened here: 
https://github.com/screener-io/screener-runner/blob/master/src/tunnel.js#L5
I think it's the syntax issue, 4.5.0 does not support this usage.

**Comment by xwa130saucer** [comment]
@vrunoa @loyalchow 
Sorry to bother you to test on different versions of node, I think this is what I should do prior to opening this PR. Could you please list the versions we need to take care and I'll test on my machine.

---------
PS: does sauce have some features to do similar tests (running stuff in different env by just clicking instead of manually switch around different node versions)? If not, we need to build this right...

File: package.json
Comment (vrunoa) [reply package.json]: @xwa130saucer `npm run postinstall --prefix` will launch the post-install script, not the sauce-connect-launcher. I can show you this on zoom tomorrow.

**Comment by vrunoa** [comment]
@xwa130saucer @loyalchow I think we should update .circleci YAML so this run in multiple nodejs versions right now is only running on node 6
https://github.com/xwa130saucer/screener-runner/blob/master/.circleci/config.yml#L5
We could also add a functional test to make sure once screener-runner is installed, the sc binary is downloaded properly on each node version.

**Comment by xwa130saucer** [comment]
@vrunoa I am handling these two points now, will release them soon. Also I need to confirm the node version with you and @loyalchow :
We only support version 6+ right?
Do we only test the `LTS` version or what other specific versions do we test?

**Comment by loyalchow** [comment]
@xwa130saucer yes, let's support 6+ only. testing on LTS versions sounds reasonable

File: package.json
Comment (xwa130saucer) [reply package.json]: @vrunoa, Sorry about the late reply, this somehow fell through the crack... 
Let's talk in person when I come back to our office about this : )

**Comment by xwa130saucer** [comment]
@vrunoa @loyalchow 
PR is ready for review. Thanks!

File: scripts/install.js
Comment (vrunoa) [inline scripts/install.js:12]: @xwa130saucer let's raise the error so this dotnset silently fail and if possible let's change the console.log to console.error

File: package.json
Comment (vrunoa) [inline package.json]: @xwa130saucer @loyalchow are we going to fork and publish a new package before merging?

File: test/tunnel.spec.js
Comment (vrunoa) [inline test/tunnel.spec.js]: @loyalchow @xwa130saucer this feels more like a functional test, what do you think on spliting tests into different folders
```
test
|_ unit
|_ functional
```

File: scripts/install.js
Comment (xwa130saucer) [reply scripts/install.js:12]: Added `process.exit(1)` and changed the first `.log` to `.error` since the second one is more likely written for users.

File: package.json
Comment (xwa130saucer) [reply package.json]: That's a good question, we could either do this or contact with Bermi to ask him if he is willing to migrate his repo to `saucelabs` github repo and maintain there? Let's discuss in the huddle?

File: test/tunnel.spec.js
Comment (xwa130saucer) [reply test/tunnel.spec.js]: I'm OK with both, how do you think? @loyalchow

File: README.md
Comment (loyalchow) [inline README.md]: Let's remove this to simplify the docs for end users, as for screener users they won't know what this is

File: package.json
Comment (loyalchow) [reply package.json]: Let's fork and publish (but in a different ticket). Migrating to sauce repo will take time - forking will allow us to move forward now

File: test/tunnel.spec.js
Comment (loyalchow) [reply test/tunnel.spec.js]: I think this is fine with the current tests. But let's separate into another spec file - `install.spec.js` or something like that.

**Review by loyalchow — CHANGES_REQUESTED** [review]

File: package.json
Comment (xwa130saucer) [reply package.json]: Do you think `npm install` directly from `github` or we give another name and publish to npm?

Comment (xwa130saucer) [reply package.json]: I addressed in this PR since I found out it's just one line change, is this OK?

File: test/tunnel.spec.js
Comment (xwa130saucer) [reply test/tunnel.spec.js]: Addressed

File: README.md
Comment (xwa130saucer) [reply README.md]: Removed

**Comment by xwa130saucer** [comment]
@vrunoa @loyalchow 
Updated, please do another round of review, thanks

**Comment by loyalchow** [comment]
Thanks @xwa130saucer, LGTM 👍

File: .circleci/config.yml
Comment (vrunoa) [inline .circleci/config.yml]: @xwa130saucer if possible let's add a line-break here

File: scripts/install.js
Comment (vrunoa) [inline scripts/install.js]: is this true? 
```console.log('sauce-connect-launcher will attempt to re-download next time it is run.');```

File: package.json
Comment (vrunoa) [inline package.json]: @xwa130saucer would this require people to have ssh keys setup up in order to install the package?

File: scripts/install.js
Comment (xwa130saucer) [reply scripts/install.js]: Technically, it should be next it is installed (`npm install`) thanks for pointing out, I will make it strict.

File: package.json
Comment (xwa130saucer) [reply package.json]: No, it doesn't. Because we host it as a public visible repo.

File: .circleci/config.yml
Comment (xwa130saucer) [reply .circleci/config.yml]: Addressed

File: scripts/install.js
Comment (xwa130saucer) [reply scripts/install.js]: changed wording

File: package.json
Comment (xwa130saucer) [reply package.json]: According to: 
https://docs.npmjs.com/cli/install
`npm install github:<githubname>/<githubrepo>[#<commit-ish>]`
we are downloading repo using https protocol, thanks for pointing out!

**Pull request merged**
